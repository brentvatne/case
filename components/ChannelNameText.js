import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text } from 'react-native'

import Colors from '../constants/Colors'
import NavigatorService from '../utilities/navigationService'

const styles = StyleSheet.create({
  text: {
    fontWeight: 'bold',
  },
  channelTitlePrivate: {
    color: Colors.private,
  },
  channelTitleClosed: {
    color: Colors.closed,
  },
  channelTitlePublic: {
    color: Colors.public,
  },
})

export default class ChannelNameText extends React.Component {
  constructor(props) {
    super(props)
    this.goToChannel = this.goToChannel.bind(this)
  }

  goToChannel() {
    this.props.onPress()
    NavigatorService.navigate('channel', { id: this.props.channel.id })
  }

  render() {
    const privacyStyle = {
      public: styles.channelTitlePublic,
      closed: styles.channelTitleClosed,
      private: styles.channelTitlePrivate,
    }[this.props.channel.visibility]

    const { channel, style } = this.props
    const textStyle = [styles.text, privacyStyle, style]

    return (
      <Text style={textStyle} onPress={this.goToChannel}>{channel.title} </Text>
    )
  }
}

ChannelNameText.propTypes = {
  style: PropTypes.any,
  channel: PropTypes.shape({
    id: PropTypes.any,
    visibility: PropTypes.any,
    title: PropTypes.string,
    slug: PropTypes.string,
  }).isRequired,
  onPress: PropTypes.func,
}

ChannelNameText.defaultProps = {
  style: {},
  onPress: () => null,
}
