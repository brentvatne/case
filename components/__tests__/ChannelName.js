import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import ChannelNameText from '../ChannelNameText'

const channel = {
  title: 'Arena',
  visibility: 'private',
}

test('renders correctly', () => {
  const tree = renderer.create(
    <ChannelNameText channel={channel} />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
