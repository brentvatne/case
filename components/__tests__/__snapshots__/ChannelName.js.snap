// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "bold",
        },
        Object {
          "color": "#B60202",
        },
        Object {},
      ]
    }
  >
    Arena
     
  </Text>
</TouchableOpacity>
`;
